flink-runtime/src/test/java/org/apache/flink/runtime/metrics/MetricRegistryImplTest.java

```diff
@@ -369,21 +367,17 @@ public void testQueryActorShutdown() throws Exception {
 
 		MetricRegistryImpl registry = new MetricRegistryImpl(MetricRegistryConfiguration.defaultMetricRegistryConfiguration());
 
-		final ActorSystem actorSystem = AkkaUtils.createDefaultActorSystem();
+		final RpcService rpcService = new TestingRpcService();
 
-		registry.startQueryService(actorSystem, null);
+		registry.startQueryService(rpcService, null);
 
-		ActorRef queryServiceActor = registry.getQueryService();
+		MetricQueryService queryService = checkNotNull(registry.getQueryService());
 
 		registry.shutdown().get();
 
-		try {
-			Await.result(actorSystem.actorSelection(queryServiceActor.path()).resolveOne(timeout), timeout);
+		queryService.getTerminationFuture().get(timeout.toMillis(), TimeUnit.MILLISECONDS);
 
-			fail("The query actor should be terminated resulting in a ActorNotFound exception.");
-		} catch (ActorNotFound e) {
-			// we expect the query actor to be shut down
-		}
+		rpcService.stopService();
```

Comment:
Isn't the registry supposed to shut this down? in any case you likely wanna do this in a finally block.

Reply:
You're right that the registry supposed to shut it down. Remove this line.